### PR TITLE
Fix gson and guava OSGi versions

### LINKED ISSUE #1267: Fix GSON not availble warning in feature.xml
image

Turns out that the Maven artifact version is `2.7`, but its OGSi version is `2.7.0`.

- `pom.xml`:
```xml
    <dependency>
      <!-- bundle: com.google.gson -->
      <groupId>com.google.code.gson</groupId>
      <artifactId>gson</artifactId>
      <version>2.7</version>
    </dependency>
```

- Excerpt from `MANIFEST.MF` of `gson`:
```yaml
Export-Package: com.google.gson;uses:="com.google.gson.reflect,com.goo
 gle.gson.stream";version="2.7.0",com.google.gson.annotations;version=
 "2.7.0",com.google.gson.reflect;version="2.7.0",com.google.gson.strea
 m;version="2.7.0"
Bundle-Name: Gson
Bundle-Version: 2.7.0
```

### PR DESCRIPTION
**Pull request by chanseokoh**
Fixes #1267.

### CODE REVIEW

**Comment by chanseokoh** [comment]
What's interesting is that, `mvn -Pide-target-platform clean package` still pulls in guava 15:

```
chanseok@ ~/workspaceE/gcloud-eclipse-tools$ find . -name '*guava*.jar'
./gcp-repo/target/products/com.google.cloud.tools.eclipse.dist/linux/gtk/x86_64/plugins/com.google.guava_20.0.0.jar
./gcp-repo/target/products/com.google.cloud.tools.eclipse.dist/linux/gtk/x86/plugins/com.google.guava_20.0.0.jar
./gcp-repo/target/products/com.google.cloud.tools.eclipse.dist/macosx/cocoa/x86_64/Eclipse.app/Contents/Eclipse/plugins/com.google.guava_20.0.0.jar
./gcp-repo/target/products/com.google.cloud.tools.eclipse.dist/win32/win32/x86_64/plugins/com.google.guava_20.0.0.jar
./gcp-repo/target/products/com.google.cloud.tools.eclipse.dist/win32/win32/x86/plugins/com.google.guava_20.0.0.jar
./gcp-repo/target/repository/plugins/com.google.guava_20.0.0.jar
./eclipse/ide-target-platform/target/repository/plugins/com.google.guava_20.0.0.jar
./eclipse/ide-target-platform/target/repository/plugins/com.google.guava_15.0.0.v201403281430.jar
chanseok@ ~/workspaceE/gcloud-eclipse-tools$ 
```

**Comment by elharo** [comment]
I think I've seen Guava 15 get pulled in somewhere recently, maybe from ide-login or appengine-plugins-core?

File: eclipse/ide-target-platform/category.xml [hunk 1]
```diff
@@ -24,8 +24,8 @@
    <bundle id="org.hamcrest" version="1.1.0.v20090501071000"/>
    <bundle id="org.mortbay.jetty.server" version="6.1.26"/>
    <bundle id="org.mortbay.jetty.util" version="6.1.26"/>
-   <bundle id="com.google.guava" version="20.0"/>
-   <bundle id="com.google.gson" version="2.7"/>
+   <bundle id="com.google.guava" version="20.0.0"/>
```
Comment (elharo) [inline eclipse/ide-target-platform/category.xml]: Why do we say 20.0 in pom.xml (and apparently must use 20.0 there, 20.0.0 doesn't work) and 20.0.0 here? Is the version in the OSGI bundle different than the maven version?

Comment (chanseokoh) [inline eclipse/ide-target-platform/category.xml]: You're right. The OSGi version and the Maven version are different, like gson and freemarker. You can check the OSGi version of guava from its `MANIFEST.MF`.

**Comment by briandealwis** [comment]
Guava 15 comes from the m2e bundles:
```
Bundle-SymbolicName: org.eclipse.m2e.core;singleton:=true
Require-Bundle: …,
 com.google.guava;bundle-version="[14.0.1,16.0.0)"
```

**Pull request merged**